objects/src/accounts/package/mod.rs

```diff
@@ -0,0 +1,303 @@
+use alloc::{string::String, vec::Vec};
+use std::{collections::BTreeSet, string::ToString};
+
+use assembly::Library;
+use semver::Version;
+use serde::{
+    de::{Error as DeError, Unexpected},
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+use thiserror::Error;
+use vm_core::utils::{Deserializable, Serializable};
+
+use super::AccountType;
+
+mod storage_entry;
+pub use storage_entry::StorageEntry;
+
+// COMPONENT PACKAGE
+// ================================================================================================
+
+/// Represents a package containing a component's configuration and its associated library.
+///
+/// The `ComponentPackage` encapsulates all necessary information to initialize and manage
+/// a component within the system. It includes the configuration details and the compiled
+/// library code required for the component's operation.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct ComponentPackage {
+    config: ComponentTemplate,
```

Comment:
nit: I'm not 100% sure about the naming here. I think of this data more like "metadata" rather than config or template.

Reply:
Renamed to `ComponentMetadata`